src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,135 @@
+        assertThat(diskPool.getDisks().get(0).getMasterMountPoint(), is("/tmp"));
+    }
+
+
+    @Test
+    public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
+        URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
+        String config = resource.toString();
+        ConfigurationAsCode.get().configure(config);
+
+        // Test ExwsStep
+        ExwsStep.DescriptorImpl globalTemplateDescriptor = ExtensionList.lookupSingleton(ExwsStep.DescriptorImpl.class);
+        List<Template> templates = globalTemplateDescriptor.getTemplates();
+
+        assertThat(templates.get(0).getLabel(), is("all"));
+        assertThat(templates.get(0).getNodeDiskPools().get(0).getDiskPoolRefId(), is("dp1"));
+        NodeDisk nodeDisk = templates.get(0).getNodeDiskPools().get(0).getNodeDisks().get(0);
+        assertThat(nodeDisk.getDiskRefId(), is("dp1refid1"));
+        assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template11"));
+        nodeDisk = templates.get(0).getNodeDiskPools().get(0).getNodeDisks().get(1);
+        assertThat(nodeDisk.getDiskRefId(), is("dp1refid2"));
+        assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template12"));
+        assertThat(templates.get(0).getNodeDiskPools().get(1).getDiskPoolRefId(), is("dp2"));
+        nodeDisk = templates.get(0).getNodeDiskPools().get(1).getNodeDisks().get(0);
+        assertThat(nodeDisk.getDiskRefId(), is("dp2refid1"));
+        assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template21"));
+    }
+
+    @Test
+    public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
+        URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
+        String config = resource.toString();
+        ConfigurationAsCode.get().configure(config);
+
+        Computer computer = r.getInstance().getComputer("");
+        Node node = computer.getNode();
+        List<NodeDiskPool> nodeDiskPools = node.getNodeProperties().get(ExternalWorkspaceProperty.class).getNodeDiskPools();
+        assertThat(nodeDiskPools.get(0).getDiskPoolRefId(), is("master-node-id"));
+        assertThat(nodeDiskPools.get(0).getNodeDisks().get(0).getDiskRefId(), is("master-node-disk"));
+        assertThat(nodeDiskPools.get(0).getNodeDisks().get(0).getNodeMountPoint(), is("/tmp/master-node"));
+    }
+
+    @Test
+    public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
+        URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
+        String config = resource.toString();
+        ConfigurationAsCode.get().configure(config);
+
+        Computer computer = r.getInstance().getComputer("");
+        Node node = computer.getNode();
+        List<NodeDiskPool> nodeDiskPools = node.getNodeProperties().get(ExternalWorkspaceProperty.class).getNodeDiskPools();
+    }
+
+    @Test
+    public void exportConfiguration() throws Exception {
+        Yaml yaml = new Yaml();
```

Comment:
Your actually not testing export

Reply:
Is this a label issue, or a unit test logic issue? What would you say this method _is_ testing?

Reply:
this is unit test logic issue 😓
See here: https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md#model-export-test

Reply:
Thanks for taking the time to clarify this, it's very useful in allowing me to work on this efficiently.